R/roc.R

```diff
@@ -0,0 +1,261 @@
+auc <- function(SE, SP){
+  TPR <- SE
+  FPR <- 1 - SP
+  dFPR <- c(diff(FPR), 0)
+  dTPR <- c(diff(TPR), 0)
+  sum(TPR * dFPR) + sum(dTPR * dFPR)/2
+}
+
+#' @title 
+#' Return a list containing sensitivity (SE) and specificity (SP) values
+#' 
+#' @name 
+#' return_SE_SP
+#'
+#' @description
+#' \code{return_SE_SP} takes in a CFA fit object or vectors of model parameter
+#' estimates and returns a list object containing SE and SP values under partial
+#' and/or strict invariance.
+#'  
+#' @param out list object
+#' 
+#' @return The output will be a list.
+#' @export
+return_SE_SP <- function(cfa_fit,
```

Comment:
When writing a function that can take both `lavaan` object or the parameter estimates as input, my suggestion is to write two functions: (a) a function that takes only the parameter estimates as inputs, and (b) a function that takes the `lavaan` object as input, extract the parameter estimates, and then call the function in (a). This may avoid having a lot of conditional statements.

Reply:
Sorry, I didn't see this comment until after fixing the function - would you like me to split the current function into two, or do you think it might be okay to keep it as a single function?